lib/fhir_models/bootstrap/model.rb

```diff
@@ -22,6 +22,13 @@ def hash
       to_hash.hash
     end
 
+    def respond_to_missing?(method_name, *)
+      (defined?(self.class::MULTIPLE_TYPES) && self.class::MULTIPLE_TYPES[method_name.to_s]) ||
+        (!@extension.nil? && !@extension.empty? && !@extension.respond_to?(:to_ary)) ||
```

Comment:
What's up with the `_to_ary` stuff?

Reply:
My understanding is that some of the xml/json related tests rely on the method -- at least running rake had tons of failures without this guard, claiming that many objects had a missing method "to_ary".  I know there are a lot of implementations of to_ary (see image), but my understanding is that this might have been intercepting a reliance on a super class too early or something, claiming the object could respond to to_ary even though it isn't handled in `method_missing`?  But I'm not very confident in that theory as the final `||` branch is just relying on super... -- wondering if maybe `method_missing` needs a final branch for relying on super/if the error that `method_missing` throws is getting in the way of relying on super (my understanding is the method handling workflow is: check object methods, check method_missing, check super)
![image](https://github.com/fhir-crucible/fhir_models/assets/139473729/7b5c8ac6-2439-49cf-a8fe-4a2cc3096f9d)

Reply:
So, I did some investigation, and I don't think we should be making an exception for `#to_ary` here. The issue is that the condition in `#respond_to_missing?` doesn't completely match the logic in `#method_missing`. Compare this branch in `#method_missing`:
```ruby
      elsif !@extension.nil? && !@extension.empty?
        ext = @extension.select do |x|
          name = x.url.tr('-', '_').split('/').last
          anchor = name.split('#').last
          (method.to_s == name || method.to_s == anchor)
        end
        unless ext.first.nil?
          return ext.first.value.nil? ? ext.first : ext.first.value
        end
```
and the corresponding condition in `#respond_to_missing?` (minus the `to_ary` part):
```ruby
(!@extension.nil? && !@extension.empty?)
```
It captures the initial condition from the `#method_missing` branch, but doesn't account for the part where it only returns if an extension with that name is found.

This is pretty ugly, but unfortunately I think the real solution is to extract the code that looks for the particular extension/modifierExtension into private methods, and use those in both `#method_missing` and `#respond_to_missing?` so that the logic is identical.

Reply:
Done in most recent push -- let me know if I should pull the "unless" logic into the method as well and change it to be like `extension_exists?` or something like that.

Reply:
No, this is what I was expecting. We can't extract the `unless` part because the return has to stay there in that method.